### PR TITLE
Fix BasisEmbedding to allow only-1 bitstrings

### PR DESCRIPTION
**Pull request by mariaschuld**
Fixes issue #1112.

### CODE REVIEW

File: .github/CHANGELOG.md [hunk 1]
```diff
@@ -95,6 +95,11 @@
 
 <h3>Bug fixes</h3>
 
+* `BasisEmbedding` does now accept inputs whose bits are all ones 
+   or all zeros. 
+  [(#1114)](https://github.com/PennyLaneAI/pennylane/pull/1114)
+
+
```
Comment (mariaschuld) [inline .github/CHANGELOG.md:102]:
```suggestion
```

**Review by josh146 — APPROVED** [review]
Thanks @mariaschuld for getting to this so quickly!

Even though this is very trivial, should we add tests for the all zeros and all ones case, just to make sure it is correctly validated? This will also help catch similar bugs in the future.

File: pennylane/templates/embeddings/basis.py [hunk 2]
```diff
@@ -51,7 +51,7 @@ def _preprocess(features, wires):
 
         features = list(qml.math.toarray(features))
 
-        if set(features) != {0, 1}:
+        if any(b not in [0, 1] for b in features):
```
[the remark this answers is not in the document]
Comment (mariaschuld) [reply pennylane/templates/embeddings/basis.py:54]: Clever!

**Comment by mariaschuld** [comment]
Good idea, added new test cases...

File: pennylane/templates/embeddings/basis.py [hunk 3]
```diff
@@ -51,7 +51,7 @@ def _preprocess(features, wires):
 
         features = list(qml.math.toarray(features))
 
-        if set(features) != {0, 1}:
+        if set(features).issubset({0, 1}):
```
Comment (antalszava) [inline pennylane/templates/embeddings/basis.py:54]:
```suggestion
        if not set(features).issubset({0, 1}):
```

Comment (josh146) [reply pennylane/templates/embeddings/basis.py]: very good catch @antalszava 🤦

**Review by josh146 — APPROVED** [review]

**Pull request merged**